gpytorch/lazy/lazy_tensor.py

```diff
@@ -1677,11 +1677,19 @@ def root_inv_decomposition(self, initial_vectors=None, test_vectors=None, method
         if self.shape[-2:].numel() == 1:
             return RootLazyTensor(1 / self.evaluate().sqrt())
 
-        # no need to do extra work if we already have a cached cholesky decomposition
-        # TODO: Resolve issues related to triangular tensor instantiation \
-        # if _is_in_cache_ignore_args(self, "cholesky"):
-        #     res = self.cholesky()
-        #     return CholLazyTensor(res).inverse()
+        # check to see if we have already run lanczos for a diagonalization
```

Comment:
we should probably check the cache for both diagonalization and for cholesky, so that we don't end up unnecessarily computing things?